.circleci/config.yml

```diff
@@ -0,0 +1,63 @@
+jobs:
+  javascript-tests:
+    executor:
+      name: node/default
+      # The Node version here must be kept in sync with that in `package.json`.
+      tag: '12.14.1'
+    steps:
+      - checkout
+      - node/install-packages:
+          # `yarn install --frozen-lockfile` is run and cache is enabled by default for this orb configuration
+          pkg-manager: yarn      
+      - run:
+          command: yarn lint
+          name: Run linting
+      - run:
+          command: yarn markdownlint
+          name: Check markdown linting
+      - run:
+          command: yarn codecov
+          name: Check code coverage changes
+      - run:
+          command: yarn test:coverage
+          name: Run Jest tests
+
+  builds:
+    environment:
+      NODE_ENV: production
+      YARN_PRODUCTION: true
+    docker:
+      - image: 'circleci/python:3.7-node'
+    steps:
+      - checkout
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - node-deps-v1-{{ .Branch }}
+      - run: pip install tox
+      - run:
+          command: tox -e heroku
+          name: Run heroku builds and release checks
+      - run:
+          command: tox -e docs
+          name: Build docs
+
+  python-tests:
+    machine: true
+    steps:
+      - checkout
+      - docker/install-docker
+      - docker/install-docker-compose
+      - run: docker-compose build
+      - run: docker-compose run backend bash -c "flake8 --show-source && black --check treeherder/ && pytest --cov --cov-report=xml tests/ --runslow"
```

Comment:
I didn't quite do this the same way that was done with travis config, which was running python tests twice via different tox commands. I looked back at Armen's previous prs but I don't completely understand why we were running pytest twice in travis. @armenzg Maybe you could clarify?

And I couldn't just use the 'tox' command here to run pytest and linting checks because there seemed to be a path issue with docker-compose, even though I could change the default path for `docker/install-docker-compose` (maybe a circleCI bug?) and if you are executing any docker commands you either need to create a separate remote docker machine container (which didn't work because *something something mounting volumes*) OR have a dedicated machine that will execute the commands, which is what ended up working. It was all very confusing!

So I extrapolated what I thought were the important bits to get this up and running. There might need to be more linting checks needed besides flake8 and black (pip check??).

Reply:
The main idea is that we could support developers who did development *outside* of the Docker container (standard Python development) and those folks who did development *within* the Docker container (which I was trying to move us away from it).

I will give you a PR with my proposal.

Reply:
> The main idea is that we could support developers who did development _outside_ of the Docker container (standard Python development) and those folks who did development _within_ the Docker container (which I was trying to move us away from it).
> 
> I will give you a PR with my proposal.

Right, but why were we running those python tests twice in Travis (in docker and outside of docker) and is it needed?

Reply:
It is not needed anymore since it was Kyle and I that were doing standard Python development outside of Docker.

I had intended to get the whole team to stop doing development inside of Docker and only do standard Python development.

You can just keep one of them as you have.